refactor: 매칭 관련 코드 리팩토링

tnemnorivnelee opened the pull request:
# 🔗 관련 이슈

Resolves: #21 

## 📝 작업 내용

- features/match 관련 컴포넌트를 전반적으로 리팩토링을 진행했습니다.

## 📷 스크린샷 (UI 작업인 경우)

X

## 💬 리뷰어에게

코드 리뷰 부탁드립니다! 

이번 코드 리뷰 과정에서 소영님의 코드를 살펴보며 제 코드에서 개선하면 좋을 부분들을 발견하고 리팩토링을 진행했습니다. 

자신의 코드를 잘 작성하는 것도 중요하지만, 팀원의 코드를 꼼꼼히 리뷰하는 과정에서도 본인의 코드를 돌아볼 수 있는 기회가 됩니다! 

코드 리뷰는 처음 시작하시면 부담스러우실 수 있습니다. 

하지만 직접 코드를 작성하고 구현하는 것만큼이나 팀원의 코드를 리뷰하는 활동에도 충분한 시간을 투자하시면 좋을 것 같습니다!

## ✅ 기본 체크리스트

- [x] 코드가 정상적으로 실행되나요? (`yarn start` 빌드 에러 없음)
- [x] **(중요) 안드로이드/iOS 시뮬레이터에서 각각 확인했나요?**
- [x] 불필요한 `console.log`나 주석을 제거했나요?
- [x] `yarn format`을 실행하여 코드 스타일을 정렬했나요?
- [x] `yarn lint`를 실행하여 린트 에러가 없나요?

## 🏗️ 구조 체크리스트 (Self-Check)

> 💡 아래 항목을 위반하면 코드 리뷰어에게 반려될 수 있습니다!

### 파일 및 폴더

- [x] **1개 기능 전용 컴포넌트**는 `features/` 안에 두었나요? (❌ `shared/ui` 금지)
- [x] **재사용 컴포넌트**만 `shared/ui`에 두었나요?
- [x] **파일이 3개 이상**일 때만 폴더로 묶었나요? (불필요한 `index.ts` 방지)

### 상태 관리 & 네이밍

- [x] **서버 데이터**는 React Query, **UI 상태**는 Zustand를 사용했나요?
- [x] **Boolean 변수**는 `is`, `has`, `should`로 시작하나요?

---

🚀 **모든 체크리스트를 확인한 후 PR을 생성하세요!**

tnemnorivnelee committed:
refactor(match): 매칭 내역 화면 구조 분리

tnemnorivnelee committed:
refactor(match): 매칭 대기 화면 구조 분리

tnemnorivnelee committed:
feat: 루트 주소 진입 시 리다이렉션 주소 변경

tnemnorivnelee committed:
refactor: 불필요한 중복 코드 제거

tnemnorivnelee committed:
Merge branch 'develop' of https://github.com/Hansei-ro/hanseiro-app into refactor/match

soyeoung1 commented:
ParticipantCard.tsx  60번째 줄에서
여기서는 isEmpty가 early return으로 처리 되어서 항상 false인 거로 이해했는데 맞을까요?
혹시 맞으면 삼항 연산자를 정리하는 게 보기에 더 간결할 거 같습니다

Yeram9286 commented on src/features/match/adapters/matchHistoryAdapter.ts:
```diff
@@ -0,0 +1,26 @@
+import { ApiMatchHistoryItem, ApiMatchHistoryResponse } from '../types/matchHistory.api';
+import { MatchHistoryItemUI } from '../types/matchHistory.ui';
+
+const formatMatchHistoryDate = (matchDate: string): string => {
```
빈 문자열 반환 대신 - - 이나 날짜정보 없음 등 문제 표시를 간단하게 하는건 어떨까용?!

tnemnorivnelee replied:
오 좋은 것 같아요!

코드 영역을 좀 더 명확하게 선택해주시면 코멘트 파악하는데 더 용이할 것 같습니다..!

tnemnorivnelee commented:
> ParticipantCard.tsx 60번째 줄에서 여기서는 isEmpty가 early return으로 처리 되어서 항상 false인 거로 이해했는데 맞을까요? 혹시 맞으면 삼항 연산자를 정리하는 게 보기에 더 간결할 거 같습니다

@soyeoung1 
확인 한번 해보겠습니다!

tnemnorivnelee committed:
Merge branch 'develop' of https://github.com/Hansei-ro/hanseiro-app into refactor/match

tnemnorivnelee commented:
> ParticipantCard.tsx 60번째 줄에서 여기서는 isEmpty가 early return으로 처리 되어서 항상 false인 거로 이해했는데 맞을까요? 혹시 맞으면 삼항 연산자를 정리하는 게 보기에 더 간결할 거 같습니다

@soyeoung1 
이미 30번째 줄에서 isEmpty 분기 처리를 해둔 상태네요,,
62행에서 isEmpty 분기 처리는 필요 없는게 맞네요,, 수정했습니다!

tnemnorivnelee committed:
refactor: 코드 가독성 개선

- ParticipantCard에서 early return 이후 불필요한 isEmpty 조건 제거
- 날짜 파싱 실패 시 빈 문자열 대신 "--" 표시

soyeoung1 reviewed (approved):
수고하셨습니다!

Yeram9286 reviewed (approved):
고생하셨습니다!

tnemnorivnelee committed:
Merge branch 'develop' of https://github.com/Hansei-ro/hanseiro-app into refactor/match
